src/Prompt.js

```diff
@@ -21,37 +21,12 @@ class Prompt extends React.Component {
     super(props)
 
     this.state = {
-      value: this.props.defaultValue,
       smileyDescription: ``,
       chosenSmiley: ``,
-      required: this.props.required
     }
-
-    this.onChange = (e) => this._onChange(e)
     this.onClick = this.onClick.bind(this)
   }
 
-  static getDerivedStateFromProps(props, state) {
-    if (props.required !== state.required) {
-      return {
-        required: props.required
-      }
-    }
-    // No state update necessary
-    return null
-  }
-
-  componentDidUpdate(prevProps, prevState) {
-    if (prevState.value !== this.state.value) {
-      this.props.onChange(this.state.value, this.state.chosenSmiley)
-    }
-  }
-
-  _onChange(e) {
-    let value = e.target.value
-    this.setState({value: value})
-  }
-
   onClick(description){
```

Comment:
You're passing the index when you call this `onClick` later, but you never use it.